dispatcher.sh: allow running scripts before shutdown

okt-viacheslav opened the pull request:
Implement a new CLI parameter for specifying which scripts need to be run after the tests have finished.

This functionality is required for configurations that need to perform some non-trivial cleanup.

Reviewed-by: Sergey Nikitin <sergey.nikitin@oktet.tech>

Testing done: tested in a cloud environment where agent hosts are automatically created and destroyed

ol-alexandra commented:
Please squash these 2 commits into one.  It definitely should not be split like this.

I have no opinion about the content on your patch, and @ol-arteman is on vacation.

ol-andrewr commented on dispatcher.sh:
```diff
@@ -661,6 +678,11 @@ process_opts()
 
+            --finish-script=* )
+                process_script_opt $1 finish-script
+                TE_FINISH_SCRIPTS="${TE_FINISH_SCRIPTS} ${script_file}"
```
I think bash array would be better here to avoid issue with spaces in path.
May be tester script should be fixed before the patch as well. Please. do.

okt-viacheslav committed:
dispatcher.sh: add missing documentation for CLI parameter

The description of the --tester-script parameter was missing from the
help message.

Signed-off-by: Viacheslav Galaktionov <viacheslav.galaktionov@oktet.tech>

okt-viacheslav committed:
dispatcher.sh: use bash arrays for Tester scripts

This is a cleaner approach that handles spaces in paths better.

Signed-off-by: Viacheslav Galaktionov <viacheslav.galaktionov@oktet.tech>

okt-viacheslav committed:
dispatcher.sh: allow running scripts before shutdown

Implement a new CLI parameter for specifying which scripts need to be run after
the tests have finished.

This functionality is required for configurations that need to perform some
non-trivial cleanup.

Signed-off-by: Viacheslav Galaktionov <viacheslav.galaktionov@oktet.tech>
